dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -247,16 +264,41 @@ private String readFileFromRepo(String fileName, String reference, GHRepository
                 if (i == 0 && folders.get(i).isEmpty()) {
                     continue;
                 }
+                // build up from the root and look for folders (potentially in the cache)
                 start.add(folders.get(i));
                 String partialPath = Joiner.on("/").join(start);
                 try {
-                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo);
-                    if (innerContent != null && innerContent.getLeft().getType().equals("symlink")) {
-                        // restart the loop to look for symbolic links pointed to by symbolic links
-                        List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
-                        List<String> sublist = folders.subList(i + 1, folders.size());
-                        newfolders.addAll(sublist);
-                        folders = newfolders;
+                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo, submoduleRedirected);
+                    if (innerContent != null) {
+                        if (innerContent.getLeft().getType().equals(SYMLINK)) {
+                            // restart the loop to look for symbolic links pointed to by symbolic links
+                            List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
+                            List<String> sublist = folders.subList(i + 1, folders.size());
+                            newfolders.addAll(sublist);
+                            folders = newfolders;
+                        } else if (innerContent.getLeft().getType().equals(SUBMODULE)) {
+                            String otherRepo = innerContent.getLeft().getGitUrl();
```

Comment:
Hmm, think I answered my own question - we don't use the right side in this function when it's a submodule. Wondering if it's better to make the right side the git URL instead of null so on this line we can call `innerContent.getRight()` which matches the pattern for symlinks above